onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -418,12 +419,16 @@ and present state are optional. Present state could appear in output even when p
             "Whether every token can only attend to previous tokens. Default value is 0.",
             AttributeProto::INT,
             static_cast<int64_t>(0))
-      .Input(0, "input", "3D input tensor with shape (batch_size, sequence_length, hidden_size), hidden_size = num_heads * head_size", "T")
+      .Attr("input_dimension_swaped",
+            "Whether input shape is (sequence_length, batch_size, hidden_size) instead of (batch_size, sequence_length, hidden_size). Default value is 0.",
```

Comment:
The shape inference function for Attention accesses input_dims[1] when doing something with past state: how does the new flag interact with that code?

Reply:
Thanks for pointing it out. So currently I don't see any models using [S,B,NH] as input that have past state. Added an assertion that past state requires input [B,S,NH]